pkg/odo/cli/apiserver/apiserver.go

```diff
@@ -0,0 +1,92 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+
+	apiserver_impl "github.com/redhat-developer/odo/pkg/apiserver-impl"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+const (
+	RecommendedCommandName = "api-server"
+)
+
+type ApiServerOptions struct {
+	clientset         *clientset.Clientset
+	apiServerPortFlag int
+}
+
+func NewApiServerOptions() *ApiServerOptions {
+	return &ApiServerOptions{}
+}
+
+var _ genericclioptions.Runnable = (*ApiServerOptions)(nil)
```

Comment:
Not sure if it is already enforced elsewhere, but we could check compliance against the other interfaces as well?

```suggestion
var _ genericclioptions.Runnable = (*ApiServerOptions)(nil)
var _ genericclioptions.SignalHandler = (*ApiServerOptions)(nil)
var _ genericclioptions.Cleanuper = (*ApiServerOptions)(nil)
```